File: src/mocks/auditLog.ts
// M11 操作审计（演示数据）

export interface AuditLog {
  id: string;
  time: string;
  user: string;
  action: string;
  target: string;
  version: string;
  detail: string;
}

export const auditLogs: AuditLog[] = [
  { id: "a1", time: "2026-08-15 14:32:08", user: "陈立群", action: "导出", target: "技术标（CG-2026-1022）", version: "v2", detail: "明标导出，校验通过，哈希 A3F2…91CE" },
  { id: "a2", time: "2026-08-15 14:20:44", user: "王建军", action: "改写接受", target: "技术标 / 2.1.2 / 施工工艺", version: "v3", detail: "接受 AI 改写：虚词→量化（问题 T4）" },
  { id: "a3", time: "2026-08-15 11:05:17", user: "赵敏", action: "AI 改写", target: "资格文件 / 财务 / 2.1", version: "v1", detail: "生成对照稿，待编写人确认（问题 T3）" },
  { id: "a4", time: "2026-08-14 09:15:30", user: "李卫东", action: "导出", target: "资格文件（CG-2026-1022）", version: "v1", detail: "导出被阻断：存在未关闭废标项" },
  { id: "a5", time: "2026-08-14 08:46:12", user: "陈立群", action: "发起预审", target: "CG-2026-1022（第 2 轮）", version: "—", detail: "全量预审，风险灯：橙" },
  { id: "a6", time: "2026-08-13 17:30:55", user: "王建军", action: "引用知识", target: "知识库 / 深基坑支护方案", version: "—", detail: "插入章节 2.1.2，防串稿扫描通过" },
  { id: "a7", time: "2026-08-13 15:12:40", user: "陈立群", action: "确认对标", target: "CG-2026-1022 对标清单", version: "v1", detail: "锁定 5 条评分规则、5 条必响应条款" },
  { id: "a8", time: "2026-08-12 10:08:22", user: "赵敏", action: "解析", target: "答疑澄清（第1号）", version: "v2", detail: "增量解析，变更条款已高亮" },
];

File: backend/app/routers/auditlog.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy import func, or_
from sqlalchemy.orm import Session

from ..audit import format_cst, week_start_naive_utc
from ..auth import get_current_user
from ..db import get_db
from ..models import AuditLog, BidDocument, KnowledgeDocument, Project, TenderDocument, User
from ..schemas import (
    AuditLogListOut,
    AuditLogOut,
    SearchDocumentHit,
    SearchMemberHit,
    SearchOut,
    SearchProjectHit,
)

router = APIRouter(prefix="/api", tags=["audit"])

VALID_ACTIONS = {"解析", "确认对标", "引用知识", "发起预审", "AI 改写", "改写接受", "导出"}


def _log_to_out(row: AuditLog) -> AuditLogOut:
    return AuditLogOut(
        id=row.id,
        time=format_cst(row.created_at),
        user=row.user_name,
        action=row.action,
        target=row.target,
        version=row.version or "—",
        detail=row.detail or "",
        result=row.result or "成功",
    )


@router.get("/audit-logs", response_model=AuditLogListOut)
def list_audit_logs(
    action: str = Query(""),
    keyword: str = Query(""),
    limit: int = Query(200, ge=1, le=500),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> AuditLogListOut:
    query = db.query(AuditLog)
    if action and action != "全部" and action in VALID_ACTIONS:
        query = query.filter(AuditLog.action == action)
    kw = keyword.strip()
    if kw:
        like = f"%{kw}%"
        query = query.filter(
            or_(
                AuditLog.target.ilike(like),
                AuditLog.user_name.ilike(like),
                AuditLog.detail.ilike(like),
            )
        )
    items = query.order_by(AuditLog.created_at.desc()).limit(limit).all()

    week_start = week_start_naive_utc()
    total = db.query(func.count(AuditLog.id)).scalar() or 0
    week_total = (
        db.query(func.count(AuditLog.id)).filter(AuditLog.created_at >= week_start).scalar() or 0
    )
    week_export = (
        db.query(func.count(AuditLog.id))
        .filter(AuditLog.created_at >= week_start, AuditLog.action == "导出")
        .scalar()
        or 0
    )
    ai_count = (
        db.query(func.count(AuditLog.id))
        .filter(AuditLog.action.in_(["AI 改写", "引用知识"]))
        .scalar()
        or 0
    )

    return AuditLogListOut(
        items=[_log_to_out(row) for row in items],
        total=total,
        weekTotal=week_total,
        weekExport=week_export,
        aiCount=ai_count,
    )


@router.get("/search", response_model=SearchOut)
def global_search(
    q: str = Query(""),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> SearchOut:
    keyword = q.strip()
    if not keyword:
        return SearchOut()
    like = f"%{keyword}%"

    projects = (
        db.query(Project)
        .filter(or_(Project.name.ilike(like), Project.code.ilike(like)))
        .order_by(Project.created_at.desc())
        .limit(8)
        .all()
    )
    members = (
        db.query(User)
        .filter(
            User.disabled.isnot(True),
            or_(User.name.ilike(like), User.email.ilike(like), User.position.ilike(like)),
        )
        .limit(8)
        .all()
    )
    knowledge_docs = (
        db.query(KnowledgeDocument)
        .filter(or_(KnowledgeDocument.title.ilike(like), KnowledgeDocument.filename.ilike(like)))
        .order_by(KnowledgeDocument.created_at.desc())
        .limit(6)
        .all()
    )
    tender_docs = (
        db.query(TenderDocument)
        .filter(TenderDocument.filename.ilike(like))
        .order_by(TenderDocument.uploaded_at.desc())
        .limit(4)
        .all()
    )
    bid_docs = (
        db.query(BidDocument)
        .filter(BidDocument.filename.ilike(like))
        .order_by(BidDocument.uploaded_at.desc())
        .limit(4)
        .all()
    )

    documents: list[SearchDocumentHit] = [
        SearchDocumentHit(
            id=d.id,
            title=d.title,
            kind="知识库",
            href="/console/knowledge",
        )
        for d in knowledge_docs
    ]
    documents.extend(
        SearchDocumentHit(
            id=d.id,
            title=d.filename,
            kind="招标文件",
            href="/console/parse",
        )
        for d in tender_docs
    )
    documents.extend(
        SearchDocumentHit(
            id=d.id,
            title=d.filename,
            kind="投标文件",
            href="/console/audit",
        )
        for d in bid_docs
    )

    return SearchOut(
        projects=[
            SearchProjectHit(id=p.id, name=p.name, code=p.code, type=p.type) for p in projects
        ],
        members=[
            SearchMemberHit(
                id=u.id,
                name=u.name,
                email=u.email,
                role=u.role or "成员",
                position=u.position or "",
            )
            for u in members
        ],
        documents=documents[:8],
    )
